src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -91,6 +91,12 @@ private static function compilePattern(Route $route, $pattern, $isHost)
         $matches = array();
         $pos = 0;
         $defaultSeparator = $isHost ? '.' : '/';
+        $useUnicode = preg_match('//u', $pattern);
+        $needsUnicode = $route->getUnicode() || ($useUnicode && preg_match('/[\x80-\xFF]/', $pattern));
```

Comment:
While this is seamless most of the time, it leaves some surprising edge cases that cannot easily be fixed. The pattern `[éè]` works out of the box, so does `[éè].`, but `[ef].` suddenly requires you to explicitly enable UTF-8 matching. Of course, this can all be explained in the documentation. I just think its a bit confusing, especially because it is a bit unclear why some escape sequences have been chosen to trigger Unicode mode and others have not.

Well, this is just my gut feeling about this. Perhaps others find it easier to understand.